### ISSUE TITLE
docker::services adding incorrect --registry_mirror parameter when trying to create services

### ISSUE THREAD

**Issue by diegolima** [puppetlabs/puppetlabs-docker#198]
## What you expected to happen?
Service gets created with provided parameters

## What happened?
The command "docker service create" is adding a --registry_mirror= after the image name, causing incorrect parameters to be sent to the container

## How to reproduce it?
Hiera file:
```
docker::services:
  blueocean:
    create: true
    service_name: blueocean
    image: jenkinsci/blueocean:1.4.2
    publish: 8180:8080
    replicas: "1"
    extra_params:
      - --mount type=volume,source=blueocean,destination=/var/jenkins_home,ro=false
```
Puppet manifest:
```
  $docker_services = hiera('docker::services', undef)
  if $docker_services {
    create_resources('docker::services', $docker_services)
  }
```

## Anything else we need to know?
The flag --registry_mirror with an empty value seems to get added after the image name when creating services even when it is not specified. Even if it was specified it should be added *before* the docker image name to have the desired effect.

## Versions:
```
# puppet --version
4.10.10

# docker --version
Docker version 17.12.1-ce, build 7390fc6

# facter os
{
  architecture => "x86_64",
  family => "RedHat",
  hardware => "x86_64",
  name => "CentOS",
  release => {
    full => "7.4.1708",
    major => "7",
    minor => "4"
  },
  selinux => {
    config_mode => "enforcing",
    current_mode => "enforcing",
    enabled => true,
    enforced => true,
    policy_version => "28"
  }
}

$ puppet module list
/etc/puppetlabs/code/environments/orquestrador_prod/modules
├── derdanne-nfs (v2.0.7)
├── puppetlabs-apt (v4.5.1)
├── puppetlabs-concat (v4.2.0)
├── puppetlabs-docker (v1.0.5)
├── puppetlabs-stdlib (v4.24.0)
└── puppetlabs-translate (v1.0.0)
/etc/puppetlabs/code/environments/common
├── files (???)
├── hierdata (???)
├── manifests (???)
└── modules (???)
/etc/puppetlabs/code/modules (no modules installed)
/opt/puppetlabs/puppet/modules (no modules installed)
/etc/puppetlabs/code/environments/common/modules
└── stahnma-epel (v1.3.0)
```

## Logs:
```
Debug: Exec[blueocean docker service create](provider=posix): Executing check 'docker service ls | grep -w blueocean'
Debug: Executing: 'docker service ls | grep -w blueocean'
Debug: Exec[blueocean docker service create](provider=posix): Executing 'docker service create --name 'blueocean' --publish '8180:8080' --replicas '1' --mount type=volume,source=blueocean,destination=/var/jenkins_home,ro=false 'jenkinsci/blueocean:1.4.2' --registry-mirror='''
Debug: Executing: 'docker service create --name 'blueocean' --publish '8180:8080' --replicas '1' --mount type=volume,source=blueocean,destination=/var/jenkins_home,ro=false 'jenkinsci/blueocean:1.4.2' --registry-mirror='''
```

**Comment by davejrt** [comment]
This is fixed in the 1.1.0 release that went out a few hours ago

**Comment by diegolima** [comment]
Upgrading to 1.1.0 worked like a charm. Thanks!
